src/main/groovy/com/codebrig/arthur/observe/structure/filter/FunctionFilter.groovy

```diff
@@ -15,9 +15,22 @@ class FunctionFilter extends StructureFilter<FunctionFilter, Void> {
     private final MultiFilter filter
 
     FunctionFilter() {
-        filter = MultiFilter.matchAll(
-                new RoleFilter("DECLARATION"), new RoleFilter("FUNCTION"),
-                new RoleFilter().reject("ARGUMENT", "RETURN", "INCOMPLETE", "BODY")
+        filter = MultiFilter.matchAny(
```

Comment:
You're going to have to revert the change to this filter and remove the tests that fail. It looks like you've added support for dynamically created functions for JavaScript/Ruby and that's not the point of this filter. In the future, there may be a `DynamicFunctionFilter` or something but this filter is not it.

If you noticed, I didn't include any dynamically created methods for Java. I could have but right now I'm just worried about static functions.